### PR TITLE
Fix binary search from infinite loop

### PR DESCRIPTION
**Pull request by volfied**
@iwehrman Please review...

I created a snippet to test this as well, and tested with: empty list, one element list with existing, one element list with an non existing, your font list with repro case, run it 0-n for ranges of n...

Will address #2438 and hopefully put out the fire.

### CODE REVIEW

File: src/js/jsx/shared/Select.jsx [hunk 1]
```diff
@@ -179,20 +179,20 @@ define(function (require, exports, module) {
             // binary search for the position
             var size = options.size,
                 low = 0,
-                high = size,
+                high = size - 1,
                 middle = Math.floor(high / 2),
```
Comment (mcilroyc) [inline src/js/jsx/shared/Select.jsx]: maybe don't need this anymore?

**Comment by mcilroyc** [comment]
for kicks, would this also work:
`_.indexOf(collection.pluck(options, "id").toArray(), key, true)`
(where the last `true` arg instructs lodash to use binary search)
maybe that's only OK if we cache the plucked array.  Or maybe I'm totally crazy.

File: src/js/jsx/shared/Select.jsx (hunk 1 above)
Comment (volfied) [inline src/js/jsx/shared/Select.jsx]: True dat.

**Comment by volfied** [comment]
Yeah without caching of that collection we would be spending time with pluck (which I think is an expensive operation).

Removed the unnecessary init @mcilroyc, thanks for reviewing.

**Comment by mcilroyc** [comment]
seems like we *could* cache that array, and use a standardized lodash search.  Anyway, merging.

**Pull request merged**